Dockerfile

```diff
@@ -1,17 +1,54 @@
-FROM ghcr.io/opensafely/base-docker
+# syntax=docker/dockerfile:1.2
+#################################################
+#
+# First build a base python image from the OS base image, so we only do this once.
```

Comment:
What's only happening once? I think this is so we can do `COPY --from=builder /opt/venv /opt/venv` later, but I don't really understand what that's saving, where.

Reply:
So, both the `builder` and the `python` images need the base dependencies installing. So we do it once, and use that as the new base, rather than doing it for both `builder` and `python`

Reply:
I have updated the comment to clairfy